pymodbus/transaction.py

```diff
@@ -436,13 +435,68 @@ def delTransaction(self, tid):
         _logger.debug("Deleting transaction %d" % tid)
         if self.transactions: self.transactions.pop(0)
 
+
+class LatestTransactionManager(ModbusTransactionManager):
```

Comment:
The `LatestTransactionManager` is same as `FifoTranscationManager` class except this line. I would suggest to derive a class from `FifoTransactionManager` and override only `delTransaction` method.


```suggestion
class LatestTransactionManager(FifoTransactionManager):
    """ Implements a transaction for a manager where the
    results is the latest transaction only. This can be used where only the
    latest transaction matters.
    """
    def delTransaction(self, tid):
        """ Removes a transaction matching the referenced tid
        As there is no tid stored, clear the entire list
        :param tid: The transaction to remove
        """
        _logger.debug("Deleting transaction %d" % tid)
        self.transactions.clear()

```

Reply:
There is one other change which is getTransaction() uses pop() instead of pop(0).

delTransaction can remain the same as FIFO if we don't mind it filling up.